### PR TITLE
merge bitcoin#15638: Pull wallet code out of libbitcoin_server

### PR DESCRIPTION
**Pull request by kittywhiskers**
Split as https://github.com/dashpay/dash/pull/4562 and https://github.com/dashpay/dash/pull/4560 (this PR) for ease of review

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -698,7 +698,7 @@ void SetupServerArgs()
     gArgs.AddArg("-acceptnonstdtxn", strprintf("Relay and mine \"non-standard\" transactions (%sdefault: %u)", "testnet/regtest only; ", !testnetChainParams->RequireStandard()), true, OptionsCategory::NODE_RELAY);
     gArgs.AddArg("-dustrelayfee=<amt>", strprintf("Fee rate (in %s/kB) used to defined dust, the value of an output such that it will cost more than its value in fees at this fee rate to spend it. (default: %s)", CURRENCY_UNIT, FormatMoney(DUST_RELAY_TX_FEE)), true, OptionsCategory::NODE_RELAY);
     gArgs.AddArg("-incrementalrelayfee=<amt>", strprintf("Fee rate (in %s/kB) used to define cost of relay, used for mempool limiting and BIP 125 replacement. (default: %s)", CURRENCY_UNIT, FormatMoney(DEFAULT_INCREMENTAL_RELAY_FEE)), true, OptionsCategory::NODE_RELAY);
-    gArgs.AddArg("-bytespersigop", strprintf("Minimum bytes per sigop in transactions we relay and mine (default: %u)", DEFAULT_BYTES_PER_SIGOP), false, OptionsCategory::NODE_RELAY);
+    gArgs.AddArg("-bytespersigop", strprintf("Equivalent bytes per sigop in transactions for relay and mining (default: %u)", DEFAULT_BYTES_PER_SIGOP), false, OptionsCategory::NODE_RELAY);
```
Comment (PastaPastaPasta) [inline src/init.cpp:702]: I guess 8365 should at best be regarded as partial, as we don't actually implement this logic of treating them as larger, we just change when we reject them... Not sure if this is okay or not

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

File: src/init.cpp (hunk 1 above)
Comment (PastaPastaPasta) [reply src/init.cpp:702]: Also arguably this shouldn't be in this PR

Comment (kittywhiskers) [reply src/init.cpp:702]: It's a prerequisite for the title backport

Comment (kittywhiskers) [reply src/init.cpp:702]: Alright, renamed it to partial and force pushed!

**Comment by PastaPastaPasta** [comment]
8365 should either be in another PR (my preference, so we can squash merge this PR), or we will need to squash all of the 15638 ONCE UDJIN REVIEWS (not my preference)

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
it looks like there is another commit in 15638 which we partially backported earlier but which still has few more things to backport/fix https://github.com/UdjinM6/dash/commit/4ddb408b99fab309d3965b66fd57f05f98625aa0

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Pull request merged**